datasafe-directory/datasafe-directory-impl/src/main/java/de/adorsys/datasafe/business/impl/profile/operations/DFSBasedProfileStorageImpl.java

```diff
@@ -85,39 +90,48 @@ public void registerPrivate(CreateUserPrivateProfile profile) {
 
     @Override
     public void deregister(UserIDAuth userID) {
-        ListRequest<UserIDAuth, AbsoluteResourceLocation<PrivateResource>> privateRequest =
-                new ListRequest<>(userID, privateProfile(userID).getPrivateStorage());
-
-        List<AbsoluteResourceLocation<PrivateResource>> privateFiles =
-                listService.list(privateRequest.getLocation()).collect(Collectors.toList());
+        @NonNull
```

Comment:
Maybe NonNull actually means checking that user exists, so that it is call to userExists? instead of NonNull

Reply:
Implemented UserExists and insert check here